RoboREPL.roboFontExt/lib/roboREPL.py

```diff
@@ -325,6 +330,57 @@ def saveTheme(self, name):
         userThemes[name] = theme
         setDefaultValue("userThemes", userThemes)
 
+    def exportSettings(self):
+        if inRoboFont:
+            exportPath = mojo.UI.PutFile(message="Export RoboREPL Settings", fileName="Settings.roboREPLSettings")
+        else:
+            exportPath = vanilla.dialogs.putFile(messageText="Export RoboREPL Settings", fileName="Settings.roboREPLSettings")
+
+        if exportPath:
+            d = dict(
+                windowWidth=int(self.windowWidth),
+                windowHeight=int(self.windowHeight),
+                fontName=str(self.fontName),
+                fontSize=int(self.fontSize),
+                colorCode=tuple(self.colorCode),
+                colorStdout=tuple(self.colorStdout),
+                colorStderr=tuple(self.colorStderr),
+                colorBackground=tuple(self.colorBackground),
+                bannerGreeting=str(self.bannerGreeting),
+                startupCode=str(self.startupCode),
+                tabString=str(self.tabString),
+                showInvisibleCharacters=bool(self.showInvisibleCharacters),
+                userThemes=dict(getDefaultValue("userThemes"))
+            )
+
+            with open(exportPath, 'wb') as f:
+                plistlib.dump(d, f)
+
+    def importSettings(self):
+        if inRoboFont:
+            importPath = mojo.UI.GetFile(message="Import RoboREPL Settings", fileTypes=["roboREPLSettings"])
+        else:
+            importPath = vanilla.dialogs.getFile(messageText="Import RoboREPL Settings", fileTypes=["roboREPLSettings"])
+
+        if importPath:
+            with open(importPath, 'rb') as f:
+                d = plistlib.load(f)
```

Comment:
You should try to load/read the json. The file could written wrongly... and a proper error message should be presented

Reply:
Ok, is this sort of what you mean?

https://github.com/typesupply/roborepl/pull/21/commits/07cea8b5415fd7ed9b5468ac3f5cc66c051131c2#diff-690ba5aad2a4f2b281851ef14955b46fR361-R391